lib/webhdfs/client_v1.rb

```diff
@@ -219,12 +249,27 @@ def touch(path, options={})
     OPT_TABLE['SETTIMES'] = ['modificationtime', 'accesstime']
     alias :settimes :touch
 
-    # def delegation_token(user, options={}) # GETDELEGATIONTOKEN
-    #   raise NotImplementedError
-    # end
-    # def renew_delegation_token(token, options={}) # RENEWDELEGATIONTOKEN
-    #   raise NotImplementedError
-    # end
+    # curl -i "http://<HOST>:<PORT>/webhdfs/v1/?op=GETDELEGATIONTOKEN&renewer=<USER>"
+    def get_kerberos_delegation_token(user, options={})
+      options = options.merge({ 'renewer' => user })
+      check_options(options, OPT_TABLE['GETDELEGATIONTOKEN'])
+      res = operate_requests('GET', '/', 'GETDELEGATIONTOKEN', options)
+      check_success_json(res, 'Token')
+      @kerberos_token_updated_at = Time.now
```

Comment:
In my strong opinion (not documented anywhere yet, though), those request methods should execute constructing/issuing requests and processing responses only. Changing internal states (like `@kerberos_token_updated_at`) should be done anywhere else.

Reply:
Do you have any reason to move this statement to this line?

Reply:
you know `@kerberos_token_updated_at` is depends on request about delegation token.
I just want to merge this feature. and also, I took [mistake](https://github.com/kzk/webhdfs/pull/39#discussion_r679654328) 
(actually, I do not want to assign `@kerberos_token_updated_at` at initialize time.)

I have no idea about spliting Executing request & Changing internal status
below code is ok? do you have any good idea?

```ruby
def get_kerberos_delegation_token_and_refresh_update_at(user, options={})
  t = get_kerberos_delegation_token(user, options)
  @kerberos_token_updated_at = Time.now  # I want to update it after get token safely
  t
end

# curl -i XXX
def get_kerberos_delegation_token(user, options={})
   ## requests 
end
```

Reply:
I think it's good enough to update `@kerberos_token_updated_at` in the `get_cached_kerberos_delegation_token` method as you did before. The timestamp is controlling when the cached token should be renewed, so that updating the timestamp should be done in the method about caching.

Reply:
ok, I'll move it to cached function and test on my environments. :)

Reply:
Super!